feat(isolated-declarations)!: Rewrite `declare module X` to `declare namespace X`

leaysgur opened the pull request:
May fix #12844

At first glance, it looked like a simple change, so I created a PR.

But sorry, I think I'm not familiar with the context, so please feel free to close if not appropriate. 😓

Dunqing commented on crates/oxc_isolated_declarations/src/declaration.rs:
```diff
@@ -134,7 +134,12 @@ impl<'a> IsolatedDeclarations<'a> {
                     decl.id.clone_in(self.ast.allocator),
                     Some(TSModuleDeclarationBody::TSModuleDeclaration(inner)),
-                    decl.kind,
+                    // Follows https://github.com/microsoft/TypeScript/pull/54134
+                    if decl.kind == TSModuleDeclarationKind::Module {
+                        TSModuleDeclarationKind::Namespace
+                    } else {
+                        decl.kind
+                    },
```
I think we can use `TSModuleDeclarationKind::Namespace` directly here, as no `global {}` would enter this method.

Dunqing commented on crates/oxc_isolated_declarations/src/declaration.rs:
```diff
@@ -144,7 +149,12 @@ impl<'a> IsolatedDeclarations<'a> {
                     decl.id.clone_in(self.ast.allocator),
                     Some(TSModuleDeclarationBody::TSModuleBlock(body)),
-                    decl.kind,
+                    // The same as above
+                    if decl.kind == TSModuleDeclarationKind::Module {
+                        TSModuleDeclarationKind::Namespace
+                    } else {
+                        decl.kind
+                    },
```
Same as above

leaysgur commented:
@Dunqing Thanks for the review~! Updated. 👍🏻

Dunqing commented:
### Merge activity

* **Aug 7, 11:54 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(isolated-declarations)!: Rewrite `declare module X` to `declare namespace X` (#12845)

May fix #12844

At first glance, it looked like a simple change, so I created a PR.

But sorry, I think I'm not familiar with the context, so please feel free to close if not appropriate. 😓
